### ISSUE TITLE
Prerendering is not working with `app.baseURL`

### ISSUE THREAD

**Issue by huang-julien** [nuxt/nuxt#22010]
### Environment

------------------------------
- Operating System: Windows_NT
- Node Version:     v18.16.1
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  pnpm@7.9.0
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

stackblitz reproduction incoming

-> fixtures
set a baseURL for the `fixtures/basic` in the nuxt repo 
run ` pnpm nuxi build .\test\fixtures\basic\`

### Describe the bug

We got logs indicating that prerender is failing

````
✔ Server built in 5809ms                                                                                                                                                                         10:33:49
✔ Generated public test/fixtures/basic/.output/public                                                                                                                                      nitro 10:33:50
ℹ Initializing prerenderer                                                                                                                                                                 nitro 10:33:50
ℹ Prerendering 3 routes                                                                                                                                                                    nitro 10:34:04
  ├─ /random/a (1699ms) (Error: [500] )                                                                                                                                                     nitro 10:34:06
  ├─ /random/b (623ms) (Error: [500] )                                                                                                                                                      nitro 10:34:06
                                                                                                                                                                                            nitro 10:34:06  
Errors prerendering:
  ├─ /random/a (500)                                                                                                                                                                        nitro 10:34:06  
  ├─ /random/b (500)                                                                                                                                                                        nitro 10:34:06  
  ├─ /random/c (500)                                                                                                                                                                        nitro 10:34:06  
                       
````

EDIT: building `test/fixtures/basic` crash due to this issue

**Comment by danielroe** [comment]
I think this is the same issue as https://github.com/nuxt/nuxt/issues/22002

**Comment by huang-julien** [comment]
I also suspected this. Will test again with a fix for #22002
